analyses/molecular-subtyping-LGAT/01-subset-files-for-LGAT.Rmd

```diff
@@ -0,0 +1,227 @@
+                    negate = TRUE),
+         # Tumors
+         sample_type == "Tumor",
+         composition == "Solid Tissue")
+
+# Write this intermediate file to the subset directory as it allows for
+# inspection
+write_tsv(lgat_specimens_df, file.path(subset_dir, "lgat_metadata.tsv"))
+
+# Filter to WGS samples
+lgat_wgs_df <- lgat_specimens_df %>%
+  dplyr::filter(experimental_strategy == "WGS") %>%
+  dplyr::select(Kids_First_Biospecimen_ID)
+```
+
+### Subset consensus maf and annotate per subtype based on SNV
+
+```{r}
+# Filter consensus mutation files for LGAT subset
+consensusMutationSubset <- consensusMutation %>%
+  # find lgat samples
+  dplyr::filter(Tumor_Sample_Barcode %in% lgat_wgs_df$Kids_First_Biospecimen_ID) %>%
+  # select tumor sample barcode
+  dplyr::select(Tumor_Sample_Barcode,
+                Hugo_Symbol,
+                HGVSp_Short,
+                DOMAINS,
+                Variant_Classification) %>% 
+  dplyr::distinct() %>%
+  # join other WGS LGAT samples
+  dplyr::full_join(lgat_wgs_df,
+            by = c("Tumor_Sample_Barcode" = "Kids_First_Biospecimen_ID")) %>%
+  dplyr::filter(
+      # get BRAF mutation status
+      # canonical mutations V600E
+      HGVSp_Short %in% snvOI$BRAF_V600E$canonical[!is.na(snvOI$BRAF_V600E$canonical)] &
+        Hugo_Symbol=="BRAF" | # OR
+      # hotspot mutations in p.600 and p.599
+      grepl(BRAF_hotspot,HGVSp_Short) &
+        Hugo_Symbol=="BRAF" &
+        Variant_Classification != "Silent" | # OR
+      # and kinase domain mutation for non-canonical mutation 
+      grepl("PF07714",DOMAINS) & 
+        Hugo_Symbol=="BRAF" & 
+        Variant_Classification != "Silent" | # OR
+    # get NF1 mutation status
+      # all mutations in MAPK genes
+      Hugo_Symbol %in% snvOI$NF1$gene & 
+        Variant_Classification %in% c("Missense_Mutation","Nonsense_Mutation") |
+    # other MAPK mutation status
+      # all mutations in MAPK genes
+      Hugo_Symbol %in% snvOI$MAPK$gene | # OR
+    # RTK mutation status
+      # all mutations in RTK genes
+      Hugo_Symbol %in% snvOI$RTK$gene | # OR
+    # get FGFR mutation status
+      # canonical mutations
+      HGVSp_Short %in% snvOI$FGFR$canonical[!is.na(snvOI$FGFR$canonical)] &
+        Hugo_Symbol=="FGFR1" | # OR
+      # hotspot mutations 
+      grepl(FGFR_hotspot,HGVSp_Short) &
+        Hugo_Symbol=="FGFR1" | # OR
+    # get IDH mutation status
+      # hostspot mutations
+      grepl(IDH_hotspot,HGVSp_Short) & 
+        Hugo_Symbol %in% snvOI$IDH$gene | # OR
+    # get H3F3A mutation status
+      # canonical mutations
+      HGVSp_Short %in% snvOI$H3F3A$canonical & Hugo_Symbol %in% "H3F3A" | # OR
+    # get HIST1H3B mutation status
+      # canonical mutations
+      HGVSp_Short %in% snvOI$HIST1H3B$canonical & Hugo_Symbol %in% "HIST1H3B" | # OR
+    # get HIST1H3C mutation status
+      # canonical mutations
+      HGVSp_Short %in% snvOI$HIST1H3C$canonical & Hugo_Symbol %in% "HIST1H3C" 
+  ) %>%
+  dplyr::distinct()
```

Comment:
Even though you have a `distinct()` here, I see duplicate rows. Eg: `BS_GRS3T427 PDGFRA`

Reply:
right distinct() is not necessary here and above.

However, I think duplicate per biospecimen~Hugo_Dymbol are expected at this step. For BS_GRS3T427 there are 2 mutations in PDGFRA (p.K385M, p.K385N)   which in `Gather status of all SNVs per biospecimen` chunk we will gather the status per subtype 1 row per biospecimen.  I can add more comments around the steps describing the process

Reply:
>For BS_GRS3T427 there are 2 mutations in PDGFRA (p.K385M, p.K385N) 

Those mutations look identical to me, unless at the base pair level, there are two different mutations at the same site which make the same amino acid?

Reply:
The genomic sites for the 2 mutations are adjacent to each other:

Hugo_Symbol | Entrez_Gene_Id | Center | NCBI_Build | Chromosome | Start_Position | End_Position | Strand | Variant_Classification | Variant_Type | Reference_Allele | Tumor_Seq_Allele1 | Tumor_Seq_Allele2 | dbSNP_RS | dbSNP_Val_Status | Tumor_Sample_Barcode | Matched_Norm_Sample_Barcode | Match_Norm_Seq_Allele1 | Match_Norm_Seq_Allele2 | Tumor_Validation_Allele1 | Tumor_Validation_Allele2 | Match_Norm_Validation_Allele1 | Match_Norm_Validation_Allele2 | Verification_Status | Validation_Status | Mutation_Status | Sequencing_Phase | Sequence_Source | Validation_Method | Score | BAM_File | Sequencer | Tumor_Sample_UUID | Matched_Norm_Sample_UUID | HGVSc | HGVSp | HGVSp_Short | Transcript_ID | Exon_Number | t_depth | t_ref_count | t_alt_count | n_depth | n_ref_count | n_alt_count | all_effects | Allele | Gene | Feature | Feature_type
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
PDGFRA | 5156 | . | GRCh38 | chr4 | 54270665 | 54270665 | + | Missense_Mutation | SNP | A | A | T | . | . | BS_GRS3T427 | BS_00QJE3E5 | A | A | . | . | . | . | . | . | . | . | . | . | . | . | . | . | . | c.1154A>T | p.Lys385Met | p.K385M | ENST00000257290 | 8/23 | 85 | 48 | 35 | 43 | 43 | 0 | PDGFRA,missense_variant,p.Lys385Met,ENST00000257290,NM_006206.4,NM_001347828.1,NM_001347829.1,NM_001347830.1;AC058822.1,intron_variant,,ENST00000507166,;PDGFRA,missense_variant,p.Lys385Met,ENST00000509490,;PDGFRA,non_coding_transcript_exon_variant,,ENST00000509092,; | T | ENSG00000134853 | ENST00000257290 | Transcript
PDGFRA | 5156 | . | GRCh38 | chr4 | 54270666 | 54270666 | + | Missense_Mutation | SNP | G | G | T | novel | . | BS_GRS3T427 | BS_00QJE3E5 | G | G | . | . | . | . | . | . | . | . | . | . | . | . | . | . | . | c.1155G>T | p.Lys385Asn | p.K385N | ENST00000257290 | 8/23 | 85 | 47 | 36 | 43 | 43 | 0 | PDGFRA,missense_variant,p.Lys385Asn,ENST00000257290,NM_006206.4,NM_001347828.1,NM_001347829.1,NM_001347830.1;AC058822.1,intron_variant,,ENST00000507166,;PDGFRA,missense_variant,p.Lys385Asn,ENST00000509490,;PDGFRA,non_coding_transcript_exon_variant,,ENST00000509092,; | T | ENSG00000134853 | ENST00000257290 | Transcript

Reply:
oh gosh, M and N! My eyes were reading them both as M. Ok, this makes sense.